spec/delete_host_spec.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require 'rails_helper' # spec_helper?
+require 'sidekiq/testing'
+
+describe DeleteHost do
+  let(:user) { FactoryBot.create(:user) }
+  let(:host) { FactoryBot.create(:host, org_id: user.account.org_id) }
+  let(:profile) { FactoryBot.create(:profile, :with_rules, account: user.account) }
+  let(:message_id) { host.id }
+  let(:message) { { 'id' => message_id, 'type' => 'delete' } }
+  let!(:test_result) { FactoryBot.create(:test_result, profile: profile, host: host) }
+  let!(:rule_result) do
+    FactoryBot.create(:rule_result, rule: profile.rules.sample, test_result: test_result, host: host)
+  end
+
+  before do
+    logger = double(Rails.logger)
+    allow(logger).to receive(:info)
+    allow(logger).to receive(:debug)
+    allow(Sidekiq).to receive(:logger).and_return(logger)
+
+    DeleteHost.clear
```

Comment:
I think that this one should go into an `after` block instead if it is supposed to clean up stuff.